code/modules/food_and_drink/cakes.dm

```diff
@@ -24,62 +28,508 @@
 /obj/item/reagent_containers/food/snacks/cake/custom
 	name = "cake"
 	desc = "a cake"
-	icon_state = "cake"
+	icon_state = "cake_batter"
 	amount = 10
 	heal_amt = 2
 	use_bite_mask = 0
-	var/icing = 0
-	var/sliced = 0
-	var/icing_color =  "#FFFFFF"
 	flags = FPRINT | TABLEPASS | NOSPLASH
+	object_flags = IGNORE_CONTEXT_CLICK_HELD
 	initial_volume = 100
+	w_class = 4.0
+	var/sliced = 0
+	var/static/list/frostingstyles = list("classic","top swirls","bottom swirls","spirals","rose spirals")
+	var/clayer = 1
+	var/amount2 //holds the amount of slices in cake 2 (not entirely, but its used for a bit of math later)
+	var/amount3 //same for cake 3
+	var/cake_candle
+	var/litfam //is the cake lit (candle)
+
+	New()
+		..()
+		contextLayout = new /datum/contextLayout/default()
+		for(var/datum/contextAction/C in src.contextActions)
+			C.dispose()
```

Comment:
excuse you?

Reply:
this is based on other parts of the code surrounding multicontext menus. i'll test it real quick to see if i can omit it